src/server/handlers.rs

```diff
@@ -32,10 +33,37 @@ impl ToJson for ErrorResponse {
 }
 
 
+/// `StatusHandler`s are used to modify the status
+pub trait StatusHandler: Send + Sync + Any {
+    /// Called after all registered sensors are read
+    fn handle(&self, status: &mut api::Status);
+}
+
+pub struct StateFromPeopleNowPresent;
+
+impl StatusHandler for StateFromPeopleNowPresent {
+    fn handle(&self, status: &mut api::Status) {
+        // Update state depending on number of people present
+        let people_now_present: Option<u64> = status.sensors.as_ref()
```

Comment:
Honestly I just moved the code from below. But if I recall correctly it's to avoid it moving out of the `Optional` because `and_then` takes its argument by value. See also https://doc.rust-lang.org/core/option/enum.Option.html#method.as_ref